src/parser/expression/window_expression.cpp

```diff
@@ -120,6 +127,7 @@ void WindowExpression::Serialize(FieldWriter &writer) const {
 	for (auto &order : orders) {
 		order.Serialize(serializer);
 	}
+	writer.WriteOptional(filter_expr);
```

Comment:
In order to not increase the storage version, this needs to be written as the last element in the window expression

Reply:
Sneaky!  I'll give that a whirl. 

I'm not clear on why this works, though? Doesn't optional mean you write a flag?

Reply:
Optional writes a boolean followed by the actual element if the boolean is true, but the actual bool must still be read. The `FieldWriter` allows us to skip fields if they are not present in the file by looking at the field count that is written as a header, see #2969. For this to work all new elements must be added at the end, however.

Reply:
Ah, that makes sense. And it worked!